src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -792,23 +812,21 @@ static void uct_rdmacm_cm_event_handler(int fd, ucs_event_set_types_t events,
     struct rdma_cm_event *event;
     int                  ret;
 
-    for (;;) {
+    do {
+        UCS_ASYNC_TRY_BLOCK(uct_rdmacm_cm_get_async(cm), return);
```

Comment:
maybe limit the number of events to process in one handler call (try 4/8/16) instead of setting async_context=NULL and locking here